scripts/helm_docs.sh

```diff
@@ -0,0 +1,548 @@
+#!/usr/bin/env bash
```

Comment:
This whole file kinda seems overkill to me. Can't you just check to ensure that helm-docs is installed and then run it? Something like:

```
#!/usr/bin/env bash -e

if ! hash helm-docs; then
    echo "Please install helm-docs to run the pre-commit hook! https://github.com/norwoodj/helm-docs#installation"
    exit 1
fi

helm-docs

Reply:
@norwoodj This is more of a pre-commit thing, and it's needed to parse the args separate from the file arguments. The BASH `getopt` function is from here => https://github.com/agriffis/pure-getopt

As an example, if I take your BASH script above (with a slight modification of `set -ex` to see debug), and run it with the following YAML file in my branch I've submitted:

```
- id: helm_docs
  args:
    - --log-level
    - debug
  description: Uses 'helm-docs' to create documentation from the Helm chart's 'values.yaml' file, and inserts the result into a corresponding 'README.md' file.
  entry: scripts/helm_docs.sh
  exclude: charts\/.*$
  files: (Chart|requirements)\.yaml$
  language: script
  name: Helm docs
  require_serial: true
```

The result is the following:

```diff
$ pre-commit try-repo --all-files --verbose '/a/local/copy/of/this/branch/helm-docs' helm_docs

[WARNING] Creating temporary repo with uncommitted changes...
===============================================================================
Using config:
===============================================================================
repos:
-   repo: /var/folders/7p/bnb37yz95ld6bnq90blpvqw80000gn/T/tmpvzauqpez/shadow-repo
    rev: 9324abff664cfe1a767a106b140610196d5fa862
    hooks:
    -   id: helm_docs
===============================================================================
[INFO] Initializing environment for /var/folders/7p/bnb37yz95ld6bnq90blpvqw80000gn/T/tmpvzauqpez/shadow-repo.
[helm_docs] Helm docs....................................................Failed
hookid: helm_docs

Files were modified by this hook. Additional output:

+ hash helm-docs
+ helm-docs --log-level debug stable/my-first-chart/requirements.yaml stable/my-other-chart/requirements.yaml stable/my-first-chart/Chart.yaml stable/my-second-chart/Chart.yaml stable/my-other-chart/Chart.yaml stable/my-third-chard/Chart.yaml
time="2019-08-08T13:11:41+02:00" level=debug msg="Ignoring directory .git"
time="2019-08-08T13:11:41+02:00" level=debug msg="Ignoring directory stable/my-other-chart"
time="2019-08-08T13:11:41+02:00" level=info msg="Found Chart directories [stable/common, stable/my-first-chart, stable/my-second-chart, stable/my-third-chard]"
time="2019-08-08T13:11:41+02:00" level=info msg="Generating README Documentation for chart stable/common"
time="2019-08-08T13:11:41+02:00" level=info msg="Generating README Documentation for chart stable/my-third-chard"
time="2019-08-08T13:11:41+02:00" level=debug msg="Did not find template file README.md.gotmpl for chart stable/common, using default template"
time="2019-08-08T13:11:41+02:00" level=debug msg="Did not find template file README.md.gotmpl for chart stable/my-third-chard, using default template"
time="2019-08-08T13:11:41+02:00" level=info msg="Generating README Documentation for chart stable/my-first-chart"
time="2019-08-08T13:11:41+02:00" level=info msg="Generating README Documentation for chart stable/my-second-chart"
time="2019-08-08T13:11:41+02:00" level=debug msg="Using template file README.md.gotmpl for chart stable/my-first-chart"
time="2019-08-08T13:11:41+02:00" level=debug msg="Using template file README.md.gotmpl for chart stable/my-second-chart"
```

As you can see, the `pre-commit` binary simply passes the args and the files found via the `files: (Chart|requirements)\.yaml$` regex together to the script in one go. BASH scripts **do**  have argument parsers, and in the end, that's what was being accomplished here.

Anyway, I think the `helm-docs` binary automatically does recursive searching so it should be fine.